pkg/providers/cloudstack/cloudstack_test.go

```diff
@@ -1656,6 +1656,66 @@ func TestSetupAndValidateForUpgradeSSHAuthorizedKeyInvalidEtcd(t *testing.T) {
 	thenErrorExpected(t, "setting up SSH keys: ssh: no key found", err)
 }
 
+func TestValidateMachineConfigsNameUniquenessSuccess(t *testing.T) {
+	ctx := context.Background()
+	clusterSpec := givenClusterSpec(t, testClusterConfigMainFilename)
+	provider := givenProvider(t)
+	prevSpec := clusterSpec.DeepCopy()
+	prevSpec.Cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name = "prev-test-cp"
+	prevSpec.Cluster.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name = "prev-test-etcd"
+	setupContext(t)
```

Comment:
do we need this?

Reply:
Similar to vsphere, can we write a function to initialize all the test components to avoid duplicate code?